cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/AbstractContext.java

```diff
@@ -191,8 +221,7 @@ protected AbstractContext(Program program, ProgramOptions programOptions, CConfi
     this.remoteClientFactory = remoteClientFactory;
     this.programRunId = program.getId().run(ProgramRunners.getRunId(programOptions));
     this.triggeringScheduleInfo = getTriggeringScheduleInfo(programOptions);
-
-    Map<String, String> runtimeArgs = new HashMap<>(programOptions.getUserArguments().asMap());
+    Map<String, String> runtimeArgs = getRuntimeArgs(programOptions);
```

Comment:
This provides access to feature flags during the run from io.cdap.cdap.api.RuntimeContext#getRuntimeArguments. But we must ensure it's available on configuration / deploy time as well, e.g. in `io.cdap.cdap.etl.api.PipelineConfigurable#configurePipeline` and probably from `io.cdap.cdap.etl.api.StageContext`